compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -1132,15 +1133,25 @@ impl EncodeContext<'a, 'tcx> {
 
     fn encode_stability(&mut self, def_id: DefId) {
         debug!("EncodeContext::encode_stability({:?})", def_id);
-        if let Some(stab) = self.tcx.lookup_stability(def_id) {
-            record!(self.tables.stability[def_id] <- stab)
+
+        // The query lookup can take a measurable amount of time in crates with many items. Check if
+        // the stability attributes are even enabled before using their queries.
+        if self.feat.staged_api || self.tcx.sess.opts.debugging_opts.force_unstable_if_unmarked {
```

Comment:
I wonder if we should optimize force_unstable_if_unmarked - we know the result of the query ahead of time if staged API is false, right?

Reply:
Is that flag used anywhere outside building libstd?

Reply:
No (well, rustc too), but speeding up compiler development if possible seems good.